### DISCUSSION TITLE
The current user has insufficient permissions to perform the requested

### DISCUSSION THREAD

**Discussion by rahul787996** [imdhemy/laravel-in-app-purchases#383] [Q&A]
i am using laravel 7.4 and  i am try  Subscription purchases this code  ...

try {  
                        Subscription::googlePlay()->id($plan->android_product_id)->token($request->txn_id)->acknowledge("YOUR PAYLOAD");

                        $productReceipt = Subscription::googlePlay()->id($plan->android_product_id)->token($request->txn_id)->get();
                        $dataa = (array)$productReceipt;
                        
                        foreach($dataa as $key => $value){   
                            $key           = str_replace("\x00*\x00","",$key);
                            $newdata[$key] = $value;
                         }

                         if($newdata['acknowledgementState'] == 1){
                            $newTransaction->response_json = $newdata;
                            $newTransaction->status = 's';
                            $newTransaction->save();
                         }else{
                            $newTransaction->response_json = $newdata;
                            $newTransaction->status = 'f';
                            $newTransaction->save();
                            return response()->json(['error' => 'Transaction is failed']);
                         }
                    }
                    catch (\Exception $e) {
                        $newTransaction->response_json = $e->getMessage();
                        $newTransaction->status = 'f';
                        $newTransaction->save();
                        return response()->json(['error' => $e->getMessage()],400);
                    } 

but coming responce this . how to solve this issue please tell me ..

{"error":"Client error: `POST https:\/\/androidpublisher.googleapis.com\/androidpublisher\/v3\/applications\/package_name \/purchases\/subscriptions\/user_quarterly\/tokens\/purchase_token:acknowledge` resulted in a `401 Unauthorized` response:\n{\n \"error\": {\n \"code\": 401,\n \"message\": \"The current user has insufficient permissions to perform the requested o (truncated...)\n"}

**Comment by ghost** [comment]
I've found a solution that worked for me in this discussion https://github.com/imdhemy/laravel-in-app-purchases/discussions/381
